### ISSUE TITLE
NVDA handling blank lines with only spaces

### ISSUE THREAD

**Issue by LittleStar-VIP** [nvaccess/nvda#13747]
### Is your feature request related to a problem? Please describe.
When I am editing some documentsprovided by sighted person, or I am editing documents that were created by me, sometimes there are lines containing only spaces.
I sometimes do not acknowledged that there were spaces in the blank lines, and typed a paragraph after those mistakenly typed spaces.

Below is a detailed illustration for the issue:

First of all, lets type a few lines into something like notepad

'''
Hello, this is the new worl.
Hi, this is the second line.
     
'''

While I am typing those lines, I mistakenly typed some spaces into the third line, which nvda says blank.

Now, I move my edit cursor up two line to do proof reading by pressing up arrows.

Oh I found that the word 'worl' is missing a 'd' so I move the cursor to the end, before the dot, add a 'd.'

Then I keep proof reading and find that the second line is perfect.

Then I press down arrow to move to the third line.

As nvda default to not reading indentations, I assume that the line is a blank line, because I heard nvda says blank.

Then I keep typing the third line.

'''
Hello, this is the new world.
Hi, this is the second line.
     Oh this is the third line.
'''

So the problem comes, I have typed those words after those mistakenly typed spaces.

I know that for blind people who use braille display will immediately pick up those spaces because they are not relying mainly on speech of nvda. Even though nvda says the line is blank, they can still determine that the line is a line containing spaces by touching braille display.

But for people like me, who do not use braille display, I am mainly relying on the audio information provided by nvda. So if I hear blank, I assumed that it is really blank and has nothing, not even spaces.

But refer to the above situation, I assumed that it is a blank line without anything, and start typing on the third line.

But in fact, it is not really a blank line, so what I have typed is after those spaces.

### Describe the solution you'd like
I don't have a good idea at this moment.
Probably instead of saying blank for lines containing spaces, nvda says space for one space in a line or # of spaces for a line containing more than one spaces?

**Comment by XLTechie** [comment]
While not identical, a version of this problem is under discussion in #13483, as 
a result of #13431.

**Comment by Brian1Gaff** [comment]
Yes its particularly an issue if somebody else made the file in haste and 
you don't hear them until its too late so to speak. Maybe  it should make a 
noise to indicate more than one space in a line of text or something like 
that to be as low verbosity as possible.

I also find some other things irritating with notepad, but these can be laid 
at Microsoft's door, like the way it inserts a control character when you 
accidentally try to delete a word as you would in Word for example
 Brian

***@***.***
Sent via blueyonder.
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users
----- Original Message ----- 
From: "LITTLE STAR☆VIP" ***@***.***>
To: "nvaccess/nvda" ***@***.***>
Cc: "Subscribed" ***@***.***>
Sent: Saturday, May 28, 2022 6:54 AM
Subject: [nvaccess/nvda] NVDA handling blank lines with only spaces (Issue 
#13747)


### Is your feature request related to a problem? Please describe.
When I am editing some documentsprovided by sighted person, or I am editing 
documents that were created by me, sometimes there are lines containing only 
spaces.
I sometimes do not acknowledged that there were spaces in the blank lines, 
and typed a paragraph after those mistakenly typed spaces.

Below is a detailed illustration for the issue:

First of all, lets type a few lines into something like notepad

'''
Hello, this is the new worl.
Hi, this is the second line.

'''

While I am typing those lines, I mistakenly typed some spaces into the third 
line, which nvda says blank.

Now, I move my edit cursor up two line to do proof reading by pressing up 
arrows.

Oh I found that the word 'worl' is missing a 'd' so I move the cursor to the 
end, before the dot, add a 'd.'

Then I keep proof reading and find that the second line is perfect.

Then I press down arrow to move to the third line.

As nvda default to not reading indentations, I assume that the line is a 
blank line, because I heard nvda says blank.

Then I keep typing the third line.

'''
Hello, this is the new world.
Hi, this is the second line.
Oh this is the third line.
'''

So the problem comes, I have typed those words after those mistakenly typed 
spaces.

I know that for blind people who use braille display will immediately pick 
up those spaces because they are not relying mainly on speech of nvda. Even 
though nvda says the line is blank, they can still determine that the line 
is a line containing spaces by touching braille display.

But for people like me, who do not use braille display, I am mainly relying 
on the audio information provided by nvda. So if I hear blank, I assumed 
that it is really blank and has nothing, not even spaces.

But refer to the above situation, I assumed that it is a blank line without 
anything, and start typing on the third line.

But in fact, it is not really a blank line, so what I have typed is after 
those spaces.


### Describe the solution you'd like
I don't have a good idea at this moment.
Probably instead of saying blank for lines containing spaces, nvda says 
space for one space in a line or # of spaces for a line containing more than 
one spaces?

### Describe alternatives you've considered

### Additional context




-- 
Reply to this email directly or view it on GitHub:
https://github.com/nvaccess/nvda/issues/13747
